### PR TITLE
feat(readiness): require min datastore schema revision in health check

### PR DESCRIPTION
**Pull request by jon-whit**
## Description
Add a minimum required datastore schema revision check to the datastore readiness check.

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: pkg/storage/sqlcommon/sqlcommon.go [hunk 1]
```diff
@@ -445,13 +447,27 @@ func ReadAuthorizationModel(ctx context.Context, dbInfo *DBInfo, store, modelID
...
+	if err != nil {
+		return storage.ReadinessStatus{}, err
+	}
+
+	if revision < build.MinimumSupportedDatastoreSchemaRevision {
+		return storage.ReadinessStatus{
+			Message: fmt.Sprintf("datastore requires migrations: at revision '%d', but requires '%d'. Run 'openfga migrate'.", revision, build.MinimumSupportedDatastoreSchemaRevision),
+			IsReady: false,
+		}, nil
+	}
```
Comment (miparnisari) [inline pkg/storage/sqlcommon/sqlcommon.go:468]: This code isn't covered. Can you add a test for this? You can copy the one I wrote here https://github.com/openfga/openfga/blob/8e36f05d7f55d3561cd298e807829b844c2b09df/cmd/migrate/migrate_test.go#L18-L48

**Review by miparnisari — CHANGES_REQUESTED** [review]

File: pkg/storage/sqlcommon/sqlcommon.go (hunk 1 above)
Comment (jon-whit) [reply pkg/storage/sqlcommon/sqlcommon.go:468]: Pushed up.

File: pkg/server/server.go [hunk 2]
```diff
@@ -902,7 +902,18 @@ func (s *Server) IsReady(ctx context.Context) (bool, error) {
...
+	status, err := s.datastore.IsReady(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	if status.IsReady {
+		return true, nil
+	}
+
+	s.logger.WarnWithContext(ctx, "datastore health check failed", zap.Any("status", status.Message))
```
Comment (miparnisari) [inline pkg/server/server.go:915]: This log is incorrect, if the health check failed, execution should have entered line 908

Comment (jon-whit) [reply pkg/server/server.go]: Updated

**Review by miparnisari — APPROVED** [review]

**Review by miparnisari — APPROVED** [review]

**Review by jpadilla — APPROVED** [review]

**Pull request merged**